frontend/components/badge-examples.tsx

```diff
@@ -29,39 +42,52 @@ const ClickableCode = styled(StyledCode)`
   cursor: pointer;
 `
 
-function Example({ baseUrl, onClick, exampleData }) {
-  const { title, example, preview, isBadgeSuggestion } = exampleData
-  const { pattern, namedParams, queryParams } = example
-  let exampleUrl
-  let previewUrl
+function Example({
+  baseUrl,
+  onClick,
+  exampleData,
+  isBadgeSuggestion,
+}: {
+  baseUrl?: string
+  onClick: (exampleData: RenderableExampleData) => void
+  exampleData: RenderableExampleData
+  isBadgeSuggestion: boolean
+}) {
+  function handleClick() {
+    onClick(exampleData)
+  }
 
+  let exampleUrl, previewUrl
   if (isBadgeSuggestion) {
-    exampleUrl = badgeUrlFromPattern({
+    const {
+      example: { pattern, namedParams, queryParams },
+    } = exampleData as SuggestionData
+    exampleUrl = previewUrl = badgeUrlFromPattern({
       baseUrl,
       pattern,
       namedParams,
       queryParams,
     })
-    previewUrl = exampleUrl
   } else {
-    const { label, message, color, style, namedLogo } = preview
+    const {
+      example: { pattern, queryParams },
+      preview: { label, message, color, style, namedLogo },
+    } = exampleData as ExampleData
     previewUrl = staticBadgeUrl({
       baseUrl,
-      label,
+      label: label || '',
       message,
       color,
       style,
       namedLogo,
     })
     exampleUrl = badgeUrlFromPath({
       path: removeRegexpFromPattern(pattern),
-      namedParams,
```

Comment:
Were these unused/unnesccessary before?

Reply:
Yea, they were ignored by that function.